Feat/migration guide (#240)
nmshd/documentation: _docs_integrate/new-versions.md

Reviewer: I don't see the new `/Revoke` of Relationships anywhere

Author: So now I have added also the changed and new Use Cases and also the Events. It's okay so? @jkoenig134

---
# Start automatic generation
permalink: integrate/new-versions
published: true
title: "New versions"
type: scenario
toc: true
properties:
  - id: SC115
  - category: Data Model
  - description:
  - customer:
  - component: integrate
  - level:
  - implementation status: DOCS ONLY
  - documentation status:
  - published:
  - link: new-versions
require:
required_by:
# End automatic generation
---

# Migration guide due to release/v5

> After merging the PR release/v5 into the main branch of the runtime, there are some backwards incompatible changes about which the Integrators will be informed transparently in the following.

Reviewer: I'd leave out this completely. When we merge the v5 branch of the docs we will not need the paragraph anymore.

## DIDs as addresses

> The address format changed from `<3-character realm><32 or 33-character base58-string>` to `did:e:<backbone-base-url>:dids:<22-character lowercase hex string>`.

Reviewer: ```suggestion
The address format changed from `<3-character realm><32 or 33-character base58-string>` to `did:e:<backbone-hostname>:dids:<22-character lowercase hex string>`.
```

This means that the property `realm` of LocalAccount was removed.

## Removal of RelationshipChanges

> RelationshipChanges were removed. The only type of RelationshipChange used so far is the CreationChange, its `content` has been moved to a new property of the relationship creation content - the `creationContent`. If this `content` has been created by responding to a [Request]({% link _docs_integrate/data-model-overview.md %}#request) of the [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate), the [`content`]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplatecontent) changes as follows:

Reviewer: ```suggestion
RelationshipChanges were removed. The only type of RelationshipChange used before was the CreationChange, its `content` has been moved to a new property of the relationship creation content - the `creationContent`. If this `content` has been created by responding to a [Request]({% link _docs_integrate/data-model-overview.md %}#request) of the [RelationshipTemplate]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplate), the [`content`]({% link _docs_integrate/data-model-overview.md %}#relationshiptemplatecontent) changes as follows:
```

## Default Attributes

- [LocalAttributes]({% link _docs_integrate/data-model-overview.md %}#localattribute) have a new property `isDefault`. This is only used for [RepositoryAttributes]({% link _docs_integrate/attribute-introduction.md %}#repositoryattributes).
- For each [IdentityAttribute]({% link _docs_integrate/attribute-introduction.md %}#identityattributes) value type exactly one of the existing Attributes is the default Attribute, i.e. its `isDefault` property is set to `"true"`, while for all others it is undefined.
- Either it will be automatically set when creating a new [RepositoryAttribute]({% link _docs_integrate/attribute-introduction.md %}#repositoryattributes) that is the first of its value type or it can be set manually via the Use Case ChangeDefaultRepositoryAttribute. However, the desired new default RepositoryAttribute may not already have a successor.
- If the default RepositoryAttribute is succeeded, the successor will become the new default Attribute. Thus, the `succeededBy` field of the default RepositoryAttribute may never be set.

> - If the default RepositoryAttribute is deleted, the newest RepositoryAttribute of that value type without a successor will automatically become the next default RepositoryAttribute of that value type, if such a RespositoryAttribute exists.

Reviewer: not important for connector users, can be removed

- If the Sender of a Request that contains a ReadAttributeRequestItem whose `query` is a [RelationshipAttributeQuery]({% link _docs_integrate/data-model-overview.md %}#relationshipattributequery) or a [ThirdPartyRelationshipAttributeQuery]({% link _docs_integrate/data-model-overview.md %}#thirdpartyrelationshipattributequery), the Recipient of the Request can only validly answer a RelationshipAttributeQuery with a new Attribute, and a ThirdPartyRelationshipAttributeQuery with an existing Attribute. Otherwise, the [error code `error.consumption.requests.invalidAcceptParameters`]({% link _docs_integrate/error-codes.md %}#error.consumption.requests.invalidAcceptParameters) arises.

> ## Restriction of the Message, RelationshipTemplate and RelationshipCreation content types ( PR still open)

Reviewer: ```suggestion
## Restriction of the Message, RelationshipTemplate and RelationshipCreation content types
```
